lab06-kaylee/server.js

```diff
@@ -0,0 +1,58 @@
+'use strict';
+
+const Client = require('./model/client.js');
+const net = require('net');
+const EE = require('events').EventEmitter;
+const ee = new EE();
+const server = module.exports = net.createServer();
+const PORT = process.env.PORT || 3000;
+const pool = [];
+
+ee.on('default', (client, string) => {
+  client.socket.write(`Not a valid command: ${string.split(' ', 1)}\n`);
+});
+
+ee.on('/all', (client, string) => {
+  pool.forEach(c => c.socket.write(`${client.nickName}: ${string}`));
+});
+
+ee.on('/nick', (client, string) => {
+  client.nickName = string.trim();
+  pool.forEach(c => c.socket.write(`You changed your nickname to: ${client.nickName}\n`));
+});
+
+ee.on('/dm', (client, string) => {
+  pool.forEach(receiver => {
+    client.socket.write(`To ${receiver.nickName}: ${string.split(' ').slice(1).join(' ')}`);
+    receiver.socket.write(`From ${client.nickName}: ${string.split(' ').slice(1).join(' ')}`);
+  });
+});
+
+let closeSocket = function(socket) {
+  pool.splice(pool.indexOf(socket), 1);
+  pool.forEach(c => c.socket.write(`${c.userName} has left the chatroom.`));
```

Comment:
In addition to the comment I wrote below about the problems with this abstraction, You also just want to additionally ``console.log``, just so you can see that message in your server terminal too.